### ISSUE TITLE
Bug: Fail safely if data import is not successful

### ISSUE THREAD

**Issue by rpruim** [senior-knights/course-schedulizer#307]
We saw an example at the workshop where an an error during data loading led to a javascript error that caused the app not to display anything.  Since things are cached, refreshing the page just led to the same issue.  We were able to resolve this by clearing the data from the browser, but it would be better to catch errors and have the app react to them.

**Comment by charkour** [comment]
Yeah, we should add [Error Boundaries](https://reactjs.org/docs/error-boundaries.html) to the app in different levels of the component tree. We should use [this package](https://www.npmjs.com/package/react-error-boundary) because it allows the app to display a fallback where we can provide a reset button.
